whatshap/cli/haplotag.py

```diff
@@ -125,12 +125,17 @@ def attempt_add_phase_information(
         is_tagged = 1
     except KeyError:
         # check if reads with same tag have been assigned
-        if alignment.has_tag("BX") and not ignore_linked_read:
-            read_clouds = bxtag_to_haplotype[alignment.get_tag("BX")]
+        if not ignore_linked_read:
+            try:
+                read_clouds = bxtag_to_haplotype[alignment.get_tag("BX")]
```

Comment:
There are two operations in this line that could lead to a KeyError (`bxtag_to_haplotype[...]` and `get_tag()`). I think it would be clearer to only do the `get_tag()` within the `try`. Perhaps with an `else:` (untested):
```
try:
    tag = alignment.get_tag("BX")
except KeyError:
    read_clouds = []
else:
    read_clouds = bxtag_to_haplotype[tag]
```

Reply:
I can be happy with that! It is also closer to how the code was originally.